File: day-1/src/main.rs
use std::{fs, ops::Div};

fn main() {
    let contents = fs::read_to_string("input.txt").expect("couldn't read file");

    let mut position: i32 = 50;
    let mut password: i32 = 0;

    for line in contents.lines() {
        let distance: i32 = line[1..].parse().unwrap();

        match line.chars().next().unwrap() {
            'R' => {
                let old_position = position;
                position += distance
            }
            'L' => {
                let old_position = position;
                position -= distance
            }
            _ => panic!("unexpected direction"),
        }

        password += position.div(100);
        position = position.rem_euclid(100);
        // if position == 0 {
        //     password += 1;
        // }
    }

    println!("password: {}", password);
}

File: day-1/js/main.js
import fs from 'fs';

function main() {
  const file = fs.readFileSync('../input.txt', 'utf-8').trim().split('\n');
  let position = 50;
  let password = 0;

  for (let line of file) {
    const distance = parseInt(line.slice(1));
    let original = position;

    switch (line[0]) {
      case 'R':
        position += distance;
        break;
      case 'L':
        position -= distance;
        break;
    }

    while (position < 0) {
      position += 100;
      if (original !== 0) {
        password += 1;
      }
      original = position;
    }

    let dejafait = false;
    while (position > 99) {
      position -= 100;
      password += 1;
      dejafait = true;
    }

    if (position === 0 && !dejafait) {
      password += 1;
    }
  }

  console.log(password);
}

main();

